### PR TITLE
Add toolchain matrix to WASM job

### PR DESCRIPTION
**Pull request by tcharding**
Recently I changed CI to use the dtolnay runner and in doing so introduced a regression (knowingly) whereby we only ran the WASM tests with the stable toolchain.

Run the WASM tests with multiple toolchains
- stable
- beta
- 1.48.0
- nightly

### CODE REVIEW

**Comment by tcharding** [comment]
Way to go @Kixunil, you picked it! Will make this into a merge candidate at some stage. Thanks

**Review by apoelstra — APPROVED** [review]
ACK 77808b7d837da2a1c99f2a7b46213efe04e5f7dd

**Comment by apoelstra** [comment]
Happy to merge if you remove "WIP" from the descirption.

**Comment by tcharding** [comment]
Done!

**Comment by apoelstra** [comment]
Hmm, apparently this is no good. https://github.com/rust-bitcoin/rust-secp256k1/actions/runs/4759831666

Oddly it did not fail until after I merged this *and* merged #604 which doesn't even touch the github CI..

**Pull request merged**
